app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java

```diff
@@ -180,6 +193,13 @@ public static GoogleAuthInfo parseUri(Uri uri) throws GoogleAuthInfoException {
         return Base32.decode(s);
     }
 
+    /**
+     * Decodes first 26 bytes, cause next bytes means UID, Pin length and checksum - not necessary
+     */
+    public static byte[] parseYandexSecret(String s) throws EncodingException {
+        return parseSecret(s.length() == YandexInfo.SECRET_LENGTH ? s : s.substring(0, YandexInfo.SECRET_LENGTH));
```

Comment:
I haven't tested this, but I expect the app will crash if the secret is shorter than 26 characters here.

Reply:
As it turned out, the length check is enough, cause from the QR code secrets longer than 26 characters should not come
So, changed it :)